api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -71,6 +78,9 @@ public void runBeforeEachTest() {
 			encounterService = Context.getEncounterService();
 			cs = Context.getConceptService();
 		}
+		if (dao == null) {
+			dao = (ProgramWorkflowDAO) applicationContext.getBean("programWorkflowDAO");
```

Comment:
Why don't you just mark the dao field as `@Autowired`?

Reply:
I did this and it worked. But i also did the same for class ProgramWorkflowServiceImpl and this causes some errors. This is the error:
https://pastebin.com/T49m4Dzg

Reply:
Since `ProgramWorkflowService` is the class being tested here, I wouldn't autowire or mock it, but instantiate it with `new`. I know Spring discourages the use of `new`, but it test cases, its fine.

Reply:
Ok, thank you!